### DISCUSSION TITLE
Knitting Learnr from the Terminal

### DISCUSSION THREAD

**Discussion by stefanopagliari** [rstudio/learnr#645] [Q&A] [answered]
Is it possible to knit a LearnR rmd from outside RStudio?
Using Rscript -e "rmarkdown::render('FILENAME.rmd')" generates a static html file. What else should I add to the command?

**Comment by gadenbuie** [comment, marked as answer]
learnr tutorials are rendered in two steps:

1. A pre-render step prepares the text of the tutorial and sets up everything that's needed for running the Shiny app that powers the tutorial. This is the output of `rmarkdown::render()`, which is an HTML intended to be run and served as an interactive application by Shiny.
2. The app step sets up a Shiny app from the pre-rendered tutorial and serves the interactive app to the user. The command for this step is `rmarkdown::run()`, which will also trigger a re-render if there isn't a prerendered HTML file or if that file is out of date.

So you can try running

```bash
Rscript -e "rmarkdown::run("tutorial.rmd")
```

to start up the learnr tutorial. This will start up the tutorial app and give you a link where you can open the tutorial in your browser. When you stop that process in your command line, the app will stop.

**Reply by stefanopagliari** [reply]
Thank you. I had to add the path of Pandoc before but it worked.
This is what worked for me:
`Rscript -e "Sys.setenv(RSTUDIO_PANDOC='/Applications/RStudio.app/Contents/MacOS/pandoc'); rmarkdown::run(RMD PATH HERE)"`

I moved much of my editing of slides (Xaringan) and tutorials (Learnr) away from RStudio towers Obsidian so it is incredibly helpful to be able to render them pretty much with one keyboard shortcut (calling the script above from within Obsidian)
